zetaclient/bitcoin/bitcoin_signer.go

```diff
@@ -97,9 +98,70 @@ func (signer *BTCSigner) GetERC20CustodyAddress() ethcommon.Address {
 	return ethcommon.Address{}
 }
 
+// AddWithdrawTxOutputs adds the outputs to the withdraw tx
+func (signer *BTCSigner) AddWithdrawTxOutputs(
+	tx *wire.MsgTx,
+	to btcutil.Address,
+	total float64,
+	amount float64,
+	nonceMark int64,
+	fees *big.Int,
+	cancelTx bool,
+) error {
+	// convert withdraw amount to satoshis
+	amountSatoshis, err := GetSatoshis(amount)
+	if err != nil {
+		return err
+	}
+
+	// calculate remaining btc (the change) to TSS self
+	remaining := total - amount
+	remainingSats, err := GetSatoshis(remaining)
+	if err != nil {
+		return err
+	}
+	remainingSats -= fees.Int64()
+	remainingSats -= nonceMark
+	if remainingSats < 0 {
+		return fmt.Errorf("remainder value is negative: %d", remainingSats)
+	} else if remainingSats == nonceMark {
+		signer.logger.Info().Msgf("adjust remainder value to avoid duplicate nonce-mark: %d", remainingSats)
+		remainingSats--
+	}
+
+	// 1st output: the nonce-mark btc to TSS self
+	tssAddrP2WPKH := signer.tssSigner.BTCAddressWitnessPubkeyHash()
```

Comment:
Just wondering if we will always keep the tss address as P2WPKH in the future.

Reply:
Good question. I don't have answer yet and it maybe depend on our needs. For instance, if we need a new TSS address that can do something P2WPKH cannot do.